### ISSUE TITLE
Improve measure invalidation and legacy Layout(s) performance

### ISSUE THREAD

**Issue by PureWeen** [dotnet/maui#24551]
### TL;DR

- Do not bubble up `MeasureInvalidated` on scrollable views and pages
- Do not synchronously measure/arrange children on legacy layout upon child measure invalidated: simply wait for the next native layout pass

### Description of Change

I executed a [speedscope.zip](https://github.com/user-attachments/files/16814038/speedscope.zip) on [davidortinau/AllTheLists](https://github.com/davidortinau/AllTheLists/blob/net9/src/Pages/LearningPage.xaml#L20) `LearningPage` using the latest `8.0.90-ci.net8.24430.1` build.

What I noticed is that #23052 `OnChildMeasureInvalidatedInternal` was showing up and taking 2% of total time (recursive calls).
![image](https://github.com/user-attachments/assets/96001b13-fb4f-4682-97c9-6aba91216961)

This revealed an interesting thing, the time consumed in there is only due to legacy layouts (which includes `ContentView`) which are triggering synchronous native measurements instead of waiting for the layout pass which is optimized as it runs only once for each invalidated view.

![image](https://github.com/user-attachments/assets/86c4b8b4-943e-4ba5-aac7-9aa6cdfc3e78)

Therefore I realized al that code is actually not needed and substantially wrong.

After these changes, the speedscope looks like this:
![image](https://github.com/user-attachments/assets/f51114ae-670a-48de-9a36-6197b67b854d)

As you can see, legacy layout calls and page calls are gone, except for the `TemplatedCell` of the collection view which is still triggering synchronous measurements.

![image](https://github.com/user-attachments/assets/f884772c-fbf6-4bfb-a4df-12d5b664379d)

**Comment by CleverSoftwarePoland** [comment]
Was there any problem or known issue why this update was not included in SR9?
Other improvements implemented in SR9 already gave some hope for better performance, but this one looks most promising.
